x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -864,18 +864,24 @@ export function updateLayerIndexPattern(
  * - All column references are valid
  * - All prerequisites are met
  */
-export function getErrorMessages(layer: IndexPatternLayer): string[] | undefined {
-  const errors: string[] = [];
-  Object.entries(layer.columns).forEach(([columnId, column]) => {
-    // If we're transitioning to another operation, check for "new" incompleteColumns rather
-    // than "old" saved operation on the layer
-    const columnFinalRef =
-      layer.incompleteColumns?.[columnId]?.operationType || column.operationType;
-    const def = operationDefinitionMap[columnFinalRef];
-    if (def.getErrorMessage) {
-      errors.push(...(def.getErrorMessage(layer, columnId) ?? []));
-    }
-  });
+export function getErrorMessages(
+  layer: IndexPatternLayer,
+  indexPattern?: IndexPattern
+): string[] | undefined {
+  const errors: string[] = Object.entries(layer.columns)
+    .flatMap(([columnId, column]) => {
+      // If we're transitioning to another operation, check for "new" incompleteColumns rather
+      // than "old" saved operation on the layer
+      const columnFinalRef =
+        layer.incompleteColumns?.[columnId]?.operationType || column.operationType;
+      const def = operationDefinitionMap[columnFinalRef];
```

Comment:
This chunk could cause a merge conflict with my other open PR, but it should be fine to merge as-is and then fix.

Reply:
But this is unchanged from `master`

Reply:
Ok, checked your PR #88916 together with this one and it works as expected in all the various scenarios. So I'll skip to refactor this and approve your so the merge will provide a fix for both issues.